arrow/src/compute/kernels/comparison.rs

```diff
@@ -2318,7 +2318,7 @@ where
 pub fn eq_dyn(left: &dyn Array, right: &dyn Array) -> Result<BooleanArray> {
     match left.data_type() {
         DataType::Dictionary(_, _) => {
-            typed_dict_compares!(left, right, |a, b| a == b)
+            typed_dict_compares!(left, right, |a, b| a == b, |a, b| !(a ^ b))
```

Comment:
I don't understand this change -- I think the `a == b` is easier to understand and I would expect that llvm would create optimized code for whatever was being compared.

If this is clippy being silly about comparing booleans perhaps we can just ignore the lint

```suggestion
            typed_dict_compares!(left, right, |a, b| a == b, |a, b| a == b)
```

Reply:
I think we can ignore it. I think clippy is somewhat confused probably when the parameters are boolea